containerd-shim-spin/src/engine.rs

```diff
@@ -0,0 +1,238 @@
+        let app_source = self.app_source()?;
+        let resolved_app_source = self.resolve_app_source(app_source.clone())?;
+        let trigger_cmd = trigger_command_for_resolved_app_source(&resolved_app_source)
+            .with_context(|| format!("Couldn't find trigger executor for {app_source:?}"))?;
+        let mut locked_app = self.load_resolved_app_source(resolved_app_source).await?;
+        self.update_locked_app(&mut locked_app); // no-op for now
+        self.run_trigger(&trigger_cmd, locked_app).await
+    }
+
+    async fn run_trigger(&self, trigger_type: &str, app: LockedApp) -> Result<()> {
+        let working_dir = PathBuf::from("/");
+        let f = match trigger_type {
+            HttpTrigger::TRIGGER_TYPE => {
+                let http_trigger: HttpTrigger = self
+                    .build_spin_trigger(working_dir, app)
+                    .await
+                    .context("failed to build spin trigger")?;
+                info!(" >>> running spin trigger");
+                http_trigger.run(spin_trigger_http::CliArgs {
+                    address: parse_addr(SPIN_ADDR).unwrap(),
+                    tls_cert: None,
+                    tls_key: None,
+                })
+            }
+            RedisTrigger::TRIGGER_TYPE => {
+                let redis_trigger: RedisTrigger = self
+                    .build_spin_trigger(working_dir, app)
+                    .await
+                    .context("failed to build spin trigger")?;
+
+                info!(" >>> running spin trigger");
+                redis_trigger.run(spin_trigger::cli::NoArgs)
+            }
+            _ => {
+                todo!("Only Http and Redis triggers are currently supported.")
+            }
+        };
+        info!(" >>> notifying main thread we are about to start");
+        f.await
+    }
+
+    async fn load_resolved_app_source(
+        &self,
+        resolved: ResolvedAppSource,
+    ) -> anyhow::Result<LockedApp> {
+        match resolved {
+            ResolvedAppSource::File { manifest_path, .. } => {
+                // TODO: This should be configurable, see https://github.com/deislabs/containerd-wasm-shims/issues/166
+                let files_mount_strategy = FilesMountStrategy::Direct;
+
+                // create a cache directory at /.cache
+                let cache_dir = PathBuf::from("/.cache");
+                env::set_var("XDG_CACHE_HOME", &cache_dir);
+
+                if !cache_dir.exists() {
+                    tokio::fs::create_dir_all(&cache_dir)
+                        .await
+                        .with_context(|| format!("failed to create {:?}", cache_dir))?;
+                }
+
+                spin_loader::from_file(&manifest_path, files_mount_strategy).await
+            }
+            ResolvedAppSource::OciRegistry { locked_app } => Ok(locked_app),
+        }
+    }
+
+    fn update_locked_app(&self, _locked_app: &mut LockedApp) {
+        // TODO: Apply --env to component environments
```

Comment:
What env variables are missing here?  Should these come from the wasi `runtimecontext`?

Reply:
Ah this was referring to #166 although this PR doesn't really implement runtime configs.

Reply:
Since its not implemented, either remove or update with a link to the issue?

Reply:
Removed